scripts/ci/testing/ci_run_airflow_testing.sh

```diff
@@ -170,5 +161,282 @@ do
+            run_single_test_type "${@}" >"${log_file}" 2>&1 &
+        pid=$!
+        echo "The process started for ${test_type}. Log file ${log_file}"
+        echo
+        test_pids[${test_type}]=${pid}
+    done
+    start_end::group_end
+}
+
+# Monitors progress of all test types running
+# TEST_TYPES - all test types to monitor
+# test_pids - map of test types to PIDs
+function monitor_test_types() {
+    start_end::group_start "Monitoring progress of running parallel tests: ${TEST_TYPES}"
+    while true;
+    do
+        sleep "${TEST_SLEEP_TIME}"
+        if ! print_all_test_types_status; then
+            break
+        fi
+    done
+    start_end::group_end
+}
+
+
+# Monitors progress of single test type running
+# $1 - test type to monitor
+# test_pids - map of test types to PIDs
+function monitor_sequential_test_type() {
+    local test_type=$1
+    start_end::group_start "Monitoring progress of running sequential test type: $1"
+    while true;
+    do
+        sleep "${TEST_SLEEP_TIME}"
+        if ! print_test_type_status "${test_type}"; then
+            break
+        fi
+    done
+    start_end::group_end
+}
+
+# Checks status for all test types running. Returns 0 if any of the tests failed (with the exception of
+# Quarantined test.
+#
+# TEST_TYPES - all test types to run
+#    test_pids - map of test types to PIDs
+#
+# output:
+#   successful_tests - array of test types that succeeded
+#   failed_tests - array of test types that failed
+function check_test_types_status() {
+    start_end::group_start "Checking status for parallel tests: ${TEST_TYPES}"
+    local test_type_exit_code="0"
+    for test_type in "${!test_pids[@]}"; do
+        local pid="${test_pids[${test_type}]}"
+        wait "${pid}"
+        local temp_exit_code=$?
+        if [[ ${temp_exit_code} == "0" ]]; then
+            successful_tests+=("${test_type}")
+        else
+            failed_tests+=("${test_type}")
+            if [[ ${test_type} != "Quarantined" ]]; then
+                test_type_exit_code=${temp_exit_code}``
+            fi
+        fi
+    done
+    start_end::group_end
+    return "${test_type_exit_code}"
+}
+
+# Outputs logs for failed test type
+# $1 test type
+function output_log_for_failed_test_type(){
+    local failed_test_type=$1
+    local log_file="${CACHE_TMP_FILE_DIR}/output-${failed_test_type}.log"
+    start_end::group_start "${COLOR_RED}Output for failed ${failed_test_type} from log: ${log_file}${COLOR_RESET}"
+    echo "${COLOR_RED}##### Test type ${failed_test_type} failed ##### ${COLOR_RESET}"
+    echo
+    cat "${log_file}"
+    echo
+    echo "${COLOR_RED}##### Test type ${failed_test_type} failed ##### ${COLOR_RESET}"
+    echo
+    start_end::group_end
+}
+
+function print_test_summary() {
+    if [[ ${#successful_tests[@]} != 0 ]]; then
+        echo
+        echo "${COLOR_GREEN}Successful test types: ${successful_tests[*]} ${COLOR_RESET}"
+        echo
```

Comment:
I think we should have the output logs shown even in case of success, at least on CI.

The reason I'd like this is I often use information about the timing, both the total run time, and the output from `--durations` to see when things have gotten slow/where possible cases for speeding up might be.

Reply:
Yeah. That's a good reason to keep it - and on CI those are folded so yeah . we can always show it no problem.